diff -ru rdma-core-22/kernel-headers/rdma/bnxt_re-abi.h rdma-core-22-new/kernel-headers/rdma/bnxt_re-abi.h
--- rdma-core-22/kernel-headers/rdma/bnxt_re-abi.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/bnxt_re-abi.h	2019-01-28 11:33:21.000000000 -0500
@@ -41,6 +41,9 @@
 #define __BNXT_RE_UVERBS_ABI_H__
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 #define BNXT_RE_ABI_VERSION	1
 
diff -ru rdma-core-22/kernel-headers/rdma/cxgb3-abi.h rdma-core-22-new/kernel-headers/rdma/cxgb3-abi.h
--- rdma-core-22/kernel-headers/rdma/cxgb3-abi.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/cxgb3-abi.h	2019-01-28 11:37:41.000000000 -0500
@@ -34,6 +34,9 @@
 #define CXGB3_ABI_USER_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 #define IWCH_UVERBS_ABI_VERSION	1
 
diff -ru rdma-core-22/kernel-headers/rdma/cxgb4-abi.h rdma-core-22-new/kernel-headers/rdma/cxgb4-abi.h
--- rdma-core-22/kernel-headers/rdma/cxgb4-abi.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/cxgb4-abi.h	2019-01-28 11:37:41.000000000 -0500
@@ -34,6 +34,9 @@
 #define CXGB4_ABI_USER_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 #define C4IW_UVERBS_ABI_VERSION	3
 
diff -ru rdma-core-22/kernel-headers/rdma/hfi/hfi1_ioctl.h rdma-core-22-new/kernel-headers/rdma/hfi/hfi1_ioctl.h
--- rdma-core-22/kernel-headers/rdma/hfi/hfi1_ioctl.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/hfi/hfi1_ioctl.h	2019-01-28 11:37:40.000000000 -0500
@@ -52,6 +52,9 @@
 #ifndef _LINUX__HFI1_IOCTL_H
 #define _LINUX__HFI1_IOCTL_H
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 /*
  * This structure is passed to the driver to tell it where
diff -ru rdma-core-22/kernel-headers/rdma/hfi/hfi1_user.h rdma-core-22-new/kernel-headers/rdma/hfi/hfi1_user.h
--- rdma-core-22/kernel-headers/rdma/hfi/hfi1_user.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/hfi/hfi1_user.h	2019-01-28 11:37:40.000000000 -0500
@@ -58,6 +58,9 @@
 #define _LINUX__HFI1_USER_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 #include <rdma/rdma_user_ioctl.h>
 
 /*
diff -ru rdma-core-22/kernel-headers/rdma/hns-abi.h rdma-core-22-new/kernel-headers/rdma/hns-abi.h
--- rdma-core-22/kernel-headers/rdma/hns-abi.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/hns-abi.h	2019-01-28 11:37:41.000000000 -0500
@@ -35,6 +35,9 @@
 #define HNS_ABI_USER_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 struct hns_roce_ib_create_cq {
 	__aligned_u64 buf_addr;
diff -ru rdma-core-22/kernel-headers/rdma/i40iw-abi.h rdma-core-22-new/kernel-headers/rdma/i40iw-abi.h
--- rdma-core-22/kernel-headers/rdma/i40iw-abi.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/i40iw-abi.h	2019-01-28 11:37:41.000000000 -0500
@@ -38,6 +38,9 @@
 #define I40IW_ABI_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 #define I40IW_ABI_VER 5
 
diff -ru rdma-core-22/kernel-headers/rdma/ib_user_cm.h rdma-core-22-new/kernel-headers/rdma/ib_user_cm.h
--- rdma-core-22/kernel-headers/rdma/ib_user_cm.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/ib_user_cm.h	2019-01-28 11:37:40.000000000 -0500
@@ -36,6 +36,9 @@
 #define IB_USER_CM_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 #include <rdma/ib_user_sa.h>
 
 #define IB_USER_CM_ABI_VERSION 5
diff -ru rdma-core-22/kernel-headers/rdma/ib_user_ioctl_verbs.h rdma-core-22-new/kernel-headers/rdma/ib_user_ioctl_verbs.h
--- rdma-core-22/kernel-headers/rdma/ib_user_ioctl_verbs.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/ib_user_ioctl_verbs.h	2019-01-28 11:37:40.000000000 -0500
@@ -35,6 +35,9 @@
 #define IB_USER_IOCTL_VERBS_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 #include <rdma/ib_user_verbs.h>
 
 #ifndef RDMA_UAPI_PTR
diff -ru rdma-core-22/kernel-headers/rdma/ib_user_mad.h rdma-core-22-new/kernel-headers/rdma/ib_user_mad.h
--- rdma-core-22/kernel-headers/rdma/ib_user_mad.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/ib_user_mad.h	2019-01-28 11:37:40.000000000 -0500
@@ -36,6 +36,9 @@
 #define IB_USER_MAD_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 #include <rdma/rdma_user_ioctl.h>
 
 /*
diff -ru rdma-core-22/kernel-headers/rdma/ib_user_sa.h rdma-core-22-new/kernel-headers/rdma/ib_user_sa.h
--- rdma-core-22/kernel-headers/rdma/ib_user_sa.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/ib_user_sa.h	2019-01-28 11:37:40.000000000 -0500
@@ -35,6 +35,9 @@
 #define IB_USER_SA_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 enum {
 	IB_PATH_GMP		= 1,
diff -ru rdma-core-22/kernel-headers/rdma/ib_user_verbs.h rdma-core-22-new/kernel-headers/rdma/ib_user_verbs.h
--- rdma-core-22/kernel-headers/rdma/ib_user_verbs.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/ib_user_verbs.h	2019-01-28 11:37:40.000000000 -0500
@@ -38,6 +38,9 @@
 #define IB_USER_VERBS_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 /*
  * Increment this value if any changes that break userspace ABI
diff -ru rdma-core-22/kernel-headers/rdma/mlx4-abi.h rdma-core-22-new/kernel-headers/rdma/mlx4-abi.h
--- rdma-core-22/kernel-headers/rdma/mlx4-abi.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/mlx4-abi.h	2019-01-28 11:37:41.000000000 -0500
@@ -36,6 +36,9 @@
 #define MLX4_ABI_USER_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 /*
  * Increment this value if any changes that break userspace ABI
diff -ru rdma-core-22/kernel-headers/rdma/mlx5-abi.h rdma-core-22-new/kernel-headers/rdma/mlx5-abi.h
--- rdma-core-22/kernel-headers/rdma/mlx5-abi.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/mlx5-abi.h	2019-01-28 11:37:41.000000000 -0500
@@ -35,6 +35,9 @@
 #define MLX5_ABI_USER_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 #include <linux/if_ether.h>	/* For ETH_ALEN. */
 #include <rdma/ib_user_ioctl_verbs.h>
 
diff -ru rdma-core-22/kernel-headers/rdma/mlx5_user_ioctl_cmds.h rdma-core-22-new/kernel-headers/rdma/mlx5_user_ioctl_cmds.h
--- rdma-core-22/kernel-headers/rdma/mlx5_user_ioctl_cmds.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/mlx5_user_ioctl_cmds.h	2019-01-28 11:37:41.000000000 -0500
@@ -34,6 +34,9 @@
 #define MLX5_USER_IOCTL_CMDS_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 #include <rdma/ib_user_ioctl_cmds.h>
 
 enum mlx5_ib_create_flow_action_attrs {
diff -ru rdma-core-22/kernel-headers/rdma/mlx5_user_ioctl_verbs.h rdma-core-22-new/kernel-headers/rdma/mlx5_user_ioctl_verbs.h
--- rdma-core-22/kernel-headers/rdma/mlx5_user_ioctl_verbs.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/mlx5_user_ioctl_verbs.h	2019-01-28 11:37:40.000000000 -0500
@@ -34,6 +34,9 @@
 #define MLX5_USER_IOCTL_VERBS_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 enum mlx5_ib_uapi_flow_action_flags {
 	MLX5_IB_UAPI_FLOW_ACTION_FLAGS_REQUIRE_METADATA	= 1 << 0,
diff -ru rdma-core-22/kernel-headers/rdma/mthca-abi.h rdma-core-22-new/kernel-headers/rdma/mthca-abi.h
--- rdma-core-22/kernel-headers/rdma/mthca-abi.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/mthca-abi.h	2019-01-28 11:37:41.000000000 -0500
@@ -36,6 +36,9 @@
 #define MTHCA_ABI_USER_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 /*
  * Increment this value if any changes that break userspace ABI
diff -ru rdma-core-22/kernel-headers/rdma/nes-abi.h rdma-core-22-new/kernel-headers/rdma/nes-abi.h
--- rdma-core-22/kernel-headers/rdma/nes-abi.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/nes-abi.h	2019-01-28 11:37:41.000000000 -0500
@@ -39,6 +39,9 @@
 #define NES_ABI_USER_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 #define NES_ABI_USERSPACE_VER 2
 #define NES_ABI_KERNEL_VER    2
diff -ru rdma-core-22/kernel-headers/rdma/ocrdma-abi.h rdma-core-22-new/kernel-headers/rdma/ocrdma-abi.h
--- rdma-core-22/kernel-headers/rdma/ocrdma-abi.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/ocrdma-abi.h	2019-01-28 11:37:40.000000000 -0500
@@ -45,6 +45,9 @@
 #define OCRDMA_ABI_USER_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 #define OCRDMA_ABI_VERSION 2
 #define OCRDMA_BE_ROCE_ABI_VERSION 1
diff -ru rdma-core-22/kernel-headers/rdma/qedr-abi.h rdma-core-22-new/kernel-headers/rdma/qedr-abi.h
--- rdma-core-22/kernel-headers/rdma/qedr-abi.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/qedr-abi.h	2019-01-28 11:37:40.000000000 -0500
@@ -34,6 +34,9 @@
 #define __QEDR_USER_H__
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 #define QEDR_ABI_VERSION		(8)
 
diff -ru rdma-core-22/kernel-headers/rdma/rdma_netlink.h rdma-core-22-new/kernel-headers/rdma/rdma_netlink.h
--- rdma-core-22/kernel-headers/rdma/rdma_netlink.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/rdma_netlink.h	2019-01-28 11:37:41.000000000 -0500
@@ -3,6 +3,9 @@
 #define _UAPI_RDMA_NETLINK_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 enum {
 	RDMA_NL_RDMA_CM = 1,
diff -ru rdma-core-22/kernel-headers/rdma/rdma_user_cm.h rdma-core-22-new/kernel-headers/rdma/rdma_user_cm.h
--- rdma-core-22/kernel-headers/rdma/rdma_user_cm.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/rdma_user_cm.h	2019-01-28 11:37:41.000000000 -0500
@@ -35,6 +35,9 @@
 #define RDMA_USER_CM_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 #include <linux/socket.h>
 #include <linux/in6.h>
 #include <rdma/ib_user_verbs.h>
diff -ru rdma-core-22/kernel-headers/rdma/rdma_user_ioctl_cmds.h rdma-core-22-new/kernel-headers/rdma/rdma_user_ioctl_cmds.h
--- rdma-core-22/kernel-headers/rdma/rdma_user_ioctl_cmds.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/rdma_user_ioctl_cmds.h	2019-01-28 11:37:41.000000000 -0500
@@ -34,6 +34,9 @@
 #define RDMA_USER_IOCTL_CMDS_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 #include <linux/ioctl.h>
 
 /* Documentation/ioctl/ioctl-number.txt */
diff -ru rdma-core-22/kernel-headers/rdma/rdma_user_rxe.h rdma-core-22-new/kernel-headers/rdma/rdma_user_rxe.h
--- rdma-core-22/kernel-headers/rdma/rdma_user_rxe.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/rdma_user_rxe.h	2019-01-28 11:37:40.000000000 -0500
@@ -35,6 +35,9 @@
 #define RDMA_USER_RXE_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 #include <linux/socket.h>
 #include <linux/in.h>
 #include <linux/in6.h>
diff -ru rdma-core-22/kernel-headers/rdma/vmw_pvrdma-abi.h rdma-core-22-new/kernel-headers/rdma/vmw_pvrdma-abi.h
--- rdma-core-22/kernel-headers/rdma/vmw_pvrdma-abi.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/kernel-headers/rdma/vmw_pvrdma-abi.h	2019-01-28 11:37:41.000000000 -0500
@@ -48,6 +48,9 @@
 #define __VMW_PVRDMA_ABI_H__
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 #define PVRDMA_UVERBS_ABI_VERSION	3		/* ABI Version. */
 #define PVRDMA_UAR_HANDLE_MASK		0x00FFFFFF	/* Bottom 24 bits. */
diff -ru rdma-core-22/libibumad/sysfs.h rdma-core-22-new/libibumad/sysfs.h
--- rdma-core-22/libibumad/sysfs.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/libibumad/sysfs.h	2019-01-28 11:37:40.000000000 -0500
@@ -35,6 +35,9 @@
 
 #include <stdint.h>
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 #include <infiniband/umad.h>
 
 extern int sys_read_string(const char *dir_name, const char *file_name, char *str, int len);
diff -ru rdma-core-22/libibumad/umad.h rdma-core-22-new/libibumad/umad.h
--- rdma-core-22/libibumad/umad.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/libibumad/umad.h	2019-01-28 11:37:40.000000000 -0500
@@ -39,6 +39,9 @@
 #include <stdlib.h>
 #include <arpa/inet.h>
 #include <linux/types.h> /* __be16, __be32 and __be64 */
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 #ifdef __cplusplus
 extern "C" {
diff -ru rdma-core-22/libibverbs/kern-abi.h rdma-core-22-new/libibverbs/kern-abi.h
--- rdma-core-22/libibverbs/kern-abi.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/libibverbs/kern-abi.h	2019-01-28 11:37:40.000000000 -0500
@@ -36,6 +36,9 @@
 #define KERN_ABI_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 #include <assert.h>
 #include <ccan/container_of.h>
 
diff -ru rdma-core-22/libibverbs/sa.h rdma-core-22-new/libibverbs/sa.h
--- rdma-core-22/libibverbs/sa.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/libibverbs/sa.h	2019-01-28 11:37:40.000000000 -0500
@@ -36,6 +36,9 @@
 
 #include <infiniband/verbs.h>
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 struct ibv_sa_path_rec {
 	/* reserved */
diff -ru rdma-core-22/libibverbs/tm_types.h rdma-core-22-new/libibverbs/tm_types.h
--- rdma-core-22/libibverbs/tm_types.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/libibverbs/tm_types.h	2019-01-28 11:37:40.000000000 -0500
@@ -34,6 +34,9 @@
 #define _TM_TYPES_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 #include <stdint.h>
 
 #ifdef __cplusplus
diff -ru rdma-core-22/libibverbs/verbs.h rdma-core-22-new/libibverbs/verbs.h
--- rdma-core-22/libibverbs/verbs.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/libibverbs/verbs.h	2019-01-28 11:37:40.000000000 -0500
@@ -42,6 +42,9 @@
 #include <errno.h>
 #include <string.h>
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 #include <stdint.h>
 #include <infiniband/verbs_api.h>
 
diff -ru rdma-core-22/librdmacm/ib.h rdma-core-22-new/librdmacm/ib.h
--- rdma-core-22/librdmacm/ib.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/librdmacm/ib.h	2019-01-28 11:37:40.000000000 -0500
@@ -34,6 +34,9 @@
 #define _RDMA_IB_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 #include <endian.h>
 #include <string.h>
 
diff -ru rdma-core-22/providers/cxgb4/t4.h rdma-core-22-new/providers/cxgb4/t4.h
--- rdma-core-22/providers/cxgb4/t4.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/providers/cxgb4/t4.h	2019-01-28 11:37:40.000000000 -0500
@@ -37,6 +37,9 @@
 #include <stdint.h>
 #include <syslog.h>
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 #include <util/compiler.h>
 #include <util/udma_barrier.h>
 #include <endian.h>
diff -ru rdma-core-22/providers/cxgb4/t4fw_api.h rdma-core-22-new/providers/cxgb4/t4fw_api.h
--- rdma-core-22/providers/cxgb4/t4fw_api.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/providers/cxgb4/t4fw_api.h	2019-01-28 11:37:40.000000000 -0500
@@ -36,6 +36,9 @@
 #define _T4FW_INTERFACE_H_
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 enum fw_retval {
 	FW_SUCCESS		= 0,	/* completed successfully */
diff -ru rdma-core-22/providers/hns/hns_roce_u_db.h rdma-core-22-new/providers/hns/hns_roce_u_db.h
--- rdma-core-22/providers/hns/hns_roce_u_db.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/providers/hns/hns_roce_u_db.h	2019-01-28 11:37:40.000000000 -0500
@@ -31,6 +31,9 @@
  */
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 #include "hns_roce_u.h"
 
diff -ru rdma-core-22/providers/hns/hns_roce_u_hw_v1.h rdma-core-22-new/providers/hns/hns_roce_u_hw_v1.h
--- rdma-core-22/providers/hns/hns_roce_u_hw_v1.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/providers/hns/hns_roce_u_hw_v1.h	2019-01-28 11:37:40.000000000 -0500
@@ -35,6 +35,9 @@
 
 #include <stdint.h>
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 #define HNS_ROCE_CQ_DB_REQ_SOL			1
 #define HNS_ROCE_CQ_DB_REQ_NEXT			0
diff -ru rdma-core-22/providers/i40iw/i40iw_osdep.h rdma-core-22-new/providers/i40iw/i40iw_osdep.h
--- rdma-core-22/providers/i40iw/i40iw_osdep.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/providers/i40iw/i40iw_osdep.h	2019-01-28 11:37:40.000000000 -0500
@@ -40,6 +40,9 @@
 #include <string.h>
 #include <util/udma_barrier.h>
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 typedef unsigned char u8;
 typedef unsigned long long u64;
 typedef unsigned int u32;
diff -ru rdma-core-22/providers/mlx4/mlx4dv.h rdma-core-22-new/providers/mlx4/mlx4dv.h
--- rdma-core-22/providers/mlx4/mlx4dv.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/providers/mlx4/mlx4dv.h	2019-01-28 11:37:40.000000000 -0500
@@ -34,6 +34,9 @@
 #define _MLX4DV_H_
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 #include <endian.h>
 #include <infiniband/verbs.h>
 
diff -ru rdma-core-22/providers/mlx5/mlx5dv.h rdma-core-22-new/providers/mlx5/mlx5dv.h
--- rdma-core-22/providers/mlx5/mlx5dv.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/providers/mlx5/mlx5dv.h	2019-01-28 11:37:40.000000000 -0500
@@ -36,6 +36,9 @@
 #include <stdio.h>
 #include <stdbool.h>
 #include <linux/types.h> /* For the __be64 type */
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 #include <sys/types.h>
 #include <endian.h>
 #if defined(__SSE3__)
diff -ru rdma-core-22/providers/qedr/common_hsi.h rdma-core-22-new/providers/qedr/common_hsi.h
--- rdma-core-22/providers/qedr/common_hsi.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/providers/qedr/common_hsi.h	2019-01-28 11:37:40.000000000 -0500
@@ -35,6 +35,9 @@
 
 #include <stdint.h>
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 /********************************/
 /* PROTOCOL COMMON FW CONSTANTS */
diff -ru rdma-core-22/providers/qedr/rdma_common.h rdma-core-22-new/providers/qedr/rdma_common.h
--- rdma-core-22/providers/qedr/rdma_common.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/providers/qedr/rdma_common.h	2019-01-28 11:37:40.000000000 -0500
@@ -34,6 +34,9 @@
 #define __RDMA_COMMON__
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 /************************/
 /* RDMA FW CONSTANTS */
diff -ru rdma-core-22/providers/vmw_pvrdma/pvrdma_ring.h rdma-core-22-new/providers/vmw_pvrdma/pvrdma_ring.h
--- rdma-core-22/providers/vmw_pvrdma/pvrdma_ring.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/providers/vmw_pvrdma/pvrdma_ring.h	2019-01-28 11:37:40.000000000 -0500
@@ -48,6 +48,9 @@
 #define __PVRDMA_RING_H__
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 
 #define PVRDMA_INVALID_IDX	-1	/* Invalid index. */
 
diff -ru rdma-core-22/srp_daemon/srp_daemon.c rdma-core-22-new/srp_daemon/srp_daemon.c
--- rdma-core-22/srp_daemon/srp_daemon.c	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/srp_daemon/srp_daemon.c	2019-01-28 11:37:40.000000000 -0500
@@ -53,6 +53,9 @@
 #include <sys/ioctl.h>
 #include <fcntl.h>
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 #include <endian.h>
 #include <errno.h>
 #include <getopt.h>
diff -ru rdma-core-22/srp_daemon/srp_daemon.h rdma-core-22-new/srp_daemon/srp_daemon.h
--- rdma-core-22/srp_daemon/srp_daemon.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/srp_daemon/srp_daemon.h	2019-01-28 11:37:40.000000000 -0500
@@ -43,6 +43,9 @@
 #include <infiniband/verbs.h>
 #include <infiniband/umad.h>
 #include <linux/types.h>	/* __be16, __be32 and __be64 */
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 #include <ccan/build_assert.h>
 
 #include "config.h"
diff -ru rdma-core-22/srp_daemon/srp_ib_types.h rdma-core-22-new/srp_daemon/srp_ib_types.h
--- rdma-core-22/srp_daemon/srp_ib_types.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/srp_daemon/srp_ib_types.h	2019-01-28 11:37:40.000000000 -0500
@@ -39,6 +39,9 @@
 #include <endian.h>
 #include <stdint.h>
 #include <linux/types.h> /* __be16, __be32 and __be64 */
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 #include <infiniband/umad.h> /* union umad_gid */
 #include <infiniband/umad_types.h>
 
diff -ru rdma-core-22/util/mmio.h rdma-core-22-new/util/mmio.h
--- rdma-core-22/util/mmio.h	2019-01-07 15:04:49.000000000 -0500
+++ rdma-core-22-new/util/mmio.h	2019-01-28 11:37:40.000000000 -0500
@@ -54,6 +54,9 @@
 #define __UTIL_MMIO_H
 
 #include <linux/types.h>
+#ifndef __aligned_u64
+#  define __aligned_u64 __u64 __attribute__((__aligned__(8)))
+#endif
 #include <stdatomic.h>
 #include <stdint.h>
 #include <stddef.h>
